pulp_rpm/app/downloaders.py

```diff
@@ -82,6 +82,9 @@ async def _run(self, extra_data=None):
         This method provides the same return object type and documented in
         :meth:`~pulpcore.plugin.download.BaseDownloader._run`.
         """
+        # Some upstream-repos (eg, Amazon) require url-encoded paths for things like "libc++"
+        # Let's make them happy
+        self.url = quote(unquote_unreserved(self.url), safe=":/")
```

Comment:
Should this go into core?

Reply:
I thought about it, and then decided not. We've not seen a problem like this with anything other than RPM, and I don't know enough to guarantee that ansible or container-registries or pypi aren't going to break if this happens. I could def be convinced otherwise, but right now I'm being conservative.